odoo/sql_db.py

```diff
@@ -358,17 +362,20 @@ def split_for_in_conditions(self, ids, size=None):
     def print_log(self):
         global sql_counter
 
-        if not _logger.isEnabledFor(logging.DEBUG):
+        sql_log = _logger.isEnabledFor(logging.DEBUG)
+        if not sql_log and not self.sql_track_table_count:
             return
         def process(type):
             sqllogs = {'from': self.sql_from_log, 'into': self.sql_into_log}
             sum = 0
             if sqllogs[type]:
                 sqllogitems = sqllogs[type].items()
-                _logger.debug("SQL LOG %s:", type)
+                if sql_log:
+                    _logger.debug("SQL LOG %s:", type)
                 for r in sorted(sqllogitems, key=lambda k: k[1]):
-                    delay = timedelta(microseconds=r[1][1])
-                    _logger.debug("table: %s: %s/%s", r[0], delay, r[1][0])
+                    if sql_log:
+                        delay = timedelta(microseconds=r[1][1])
+                        _logger.debug("table: %s: %s/%s", r[0], delay, r[1][0])
```

Comment:
Both `if sql_log` are overkill here.  Simply call `_logger.debug`, it will simply not do anything if not enabled.